### ISSUE TITLE
[BALANCE] Stealing Chest Alerts Guards

### ISSUE THREAD

**Issue by cambragol** [cambragol/advent-of-the-trinity#268]
Wouldn't it be better if taking a chest also alerted the guards. Sure we have the virtues punishment, but this is just a point of realism. I was always disappointed when I went into Lord British's castle, and raided his treasure room, eliciting zero response from the 5 or so guards in the room.

It is a one line addition, as I already have a new 'guard alert' function to handle transgressions.

Additionally, this would also anger towns people, making their comments to you rougher, and their likelihood of turning away or even attacking you, higher. This is already built in as well.

Further, I have two types of chests, one type of which will NOT alert guards, and is 'okay' to take. These exist in 'combat' type town encounters, and in some regular towns, where someone has indicated in a conversation that the gold is free to be taken.

Any reason it would be bad to have guards alert when stealing gold?

**Comment by Fenyx4** [comment]
I feel this subverts some of RG's original design choices.

In U3 it had guards being alerted. So why would he remove that in U4? Lots of people steal gold in U3. Guards coming to kill them wasn't a deterrent. So people coming to U4 will initially steal a bunch of stuff and rejoice that they don't have to deal withe guards and steal even more... And then find out later that their actions have consequences they didn't think the game was keeping track of.

**Comment by cambragol** [comment]
A good argument. 

However, I still lean towards implementing this. The original consequences remain, if, upon stealing the treasure, they are attacked, whereupon they flee, and discover that the guards disappear. They are able to take all of the treasure, with zero consequences. Until later, when they discover a number of virtues (not just one) are very low because their theft of the gold (and subsequent fleeing from the guards) had consequences.

I have also seen this used as a quick start 'cheaty' approach by numerous players. I.e. Rather than go out and earn their initial gold the honest way (killin' evil creatures), they simply steal enough gold to arm up themselves and/or their party to a sufficiently badass level. Then restore their lost honesty/justice/honor via various cheaty methods. Which is very much subverting RG's original design choices.

However...I do lean towards there being such options. A redemptive arc etc., or an Ends Justifies the Means approach.

I wonder what Mr. Knitster thinks?

**Comment by cambragol** [comment]
A random activation could be fun too. You can get away with stealing sometimes, because, for whatever reason, the guards were not alerted. Steal too often and eventually they are alerted.

A mixture between the two arguments for/against, and a little more realistic.

**Comment by cambragol** [comment]
Another fun, and code-wise cheap option, is to have the guards only alerted if they *see* you. I.e. if they are visible, then so are you, so they see you steal and are alerted. If no guards are around, the only punishment is karma.

**Comment by MrKnitster** [comment]
I agree that stealing should be punished immediately by guards as well as by a karma hit. If the towns and guards treated the player as hostile once he has stolen in U3, why on earth did that go away in U4? Must have been a karma-trap set up to catch players.

I like the idea of guards only "catching" a player if they are on screen; i.e., visible. This feels like an appropriate technical upgrade from U3.

**Comment by cambragol** [comment]
I have been testing this in game, and much of the time it retains the original 'gameplay', as the guards are often not in the treasure room. When they do come in, occasionally, they see you steal it and become hostile. 

It feels good, as it adds some mystery as to how the guards are reacting, a tiny bit of that original mystery we all felt when playing the game, which LB capitalised on to make much of it work. He made us imagine that the game was much smarter than it was, so that we would believe in the virtues system. I think this change holds true to that original vision, and enhances it.

**Comment by cambragol** [comment]
This is now in 13.9.

Guards will go to alert mode if they see you stealing a chest. So sometimes, if they are on the screen, stealing a chest will alert them.

If they are not on the screen, they should not be alerted.

**Comment by cambragol** [comment]
tested in Castle Britannia. If guards are in the room with me when I take a chest, they are alerted. If they are out of the room, they are not alerted.
